lib/opam_cmd.ml

```diff
@@ -125,27 +125,100 @@ let classify_package ~package ~dev_repo ~archive ~pins () =
             | false -> (`Unknown host, tag) )
           | None -> err "dev-repo without host" ) )
 
-let check_if_dune ~root package =
-  Exec.get_opam_depends ~root (string_of_package package)
-  >>| List.exists (fun l -> l = "jbuilder" || l = "dune")
+let extract_opam_value = function
+  | None -> None
+  | Some str ->
+    let res =
+      try
+        Ok (OpamParser.value_from_string str "")
+      with | OpamLexer.Error _ ->
+        let error_message = Printf.sprintf "Failed to parse %S" str in
+        Error (`Msg error_message)
+    in
+    Some res
 
-let get_opam_info ~root ~pins package =
-  Exec.get_opam_dev_repo ~root (string_of_package package) >>= fun dev_repo ->
-  Exec.get_opam_archive_url ~root (string_of_package package) >>= fun archive ->
-  let dev_repo, tag = classify_package ~package ~dev_repo ~archive ~pins () in
-  check_if_dune ~root package >>= fun is_dune ->
-  Logs.info (fun l ->
+let parse_string ~field ~package v =
```

Comment:
Why not move this to `Opam_show_result` ? (Because of the dependency on `OpamParser` ?)

Reply:
`Opam_show_result` doesn't need to know how fields should be parsed so I don't think it's necessary to move this.

Reply:
The main reason why we did this was to avoid carrying dummy OpamValue with irrelevant locations around. That made testing harder and indeed, as @TheLortex said, `Opam_show_result` is just an abstraction above the `opam show` output, it's not strictly required to parse everything in there.

Reply:
I understand.
What about moving this functions as is ? Also, it could call the `get` function as it has the `package` and `field` values.